Bottom: b771c5087c3960f2a005604de299c00bb0d7b41c
Top:    4995d627381461651f8b7f364ece92f84b01dc00
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-08-16 09:17:44 -0500

Add environment variables for parameters


---
diff --git a/docker/buildbot-master/Dockerfile b/docker/buildbot-master/Dockerfile
index fff22446..49a5f4ab 100644
--- a/docker/buildbot-master/Dockerfile
+++ b/docker/buildbot-master/Dockerfile
@@ -19,9 +19,10 @@ RUN pip --no-cache-dir install \
 
 # Create buildbot user
 ARG BUILDBOT_UID=1000
+ARG BUILDBOT_DATA="/buildbot"
 COPY buildbot/ /home/buildbot/
 RUN adduser -h "/home/buildbot" -s "/bin/sh" -D -u ${BUILDBOT_UID} buildbot \
-&&  chown -v -R buildbot:buildbot "/buildbot" \
+&&  chown -v -R buildbot:buildbot "${BUILDBOT_DATA}" \
 &&  chown -v -R buildbot:buildbot "/home/buildbot"
 
 USER buildbot
diff --git a/docker/buildbot-master/buildbot/buildbot.tac b/docker/buildbot-master/buildbot/buildbot.tac
index c3b2e879..4fbddaa2 100644
--- a/docker/buildbot-master/buildbot/buildbot.tac
+++ b/docker/buildbot-master/buildbot/buildbot.tac
@@ -1,4 +1,5 @@
 import sys
+import os
 
 from twisted.application import service
 from twisted.python.log import FileLogObserver
@@ -6,7 +7,7 @@ from twisted.python.log import ILogObserver
 
 from buildbot.master import BuildMaster
 
-basedir = '/buildbot'
+basedir = os.getenv('BUILDBOT_DATA', '/buildbot')
 configfile = 'master.cfg'
 
 # note: this line is matched against to check that this is a buildmaster
diff --git a/docker/buildbot-master/buildbot/start.sh b/docker/buildbot-master/buildbot/start.sh
index a30fbf6d..5e0eed75 100755
--- a/docker/buildbot-master/buildbot/start.sh
+++ b/docker/buildbot-master/buildbot/start.sh
@@ -1,10 +1,12 @@
 #!/bin/sh
 
-mkdir -p $HOME/.ssh
-cp -R /run/secrets/ssh/* $HOME/.ssh
-chmod -R 600 $HOME/.ssh/*
+BUILDBOT_DATA=${BUILDBOT_DATA-"/buildbot"}
 
-ln -v -s -f /home/buildbot/buildbot.tac /home/buildbot/master.cfg /home/buildbot/*.py /buildbot
+mkdir -p ${HOME}/.ssh
+cp -R /run/secrets/ssh/* ${HOME}/.ssh
+chmod -R 600 ${HOME}/.ssh/*
+
+ln -v -s -f ${HOME}/buildbot.tac ${HOME}/master.cfg ${HOME}/*.py $BUILDBOT_DATA
 
 /usr/bin/buildbot --verbose upgrade-master
 exec twistd --nodaemon --logfile=- --pidfile=/tmp/twistd.pid --python=buildbot.tac
diff --git a/roles/buildbot-master/defaults/main.yml b/roles/buildbot-master/defaults/main.yml
index c40f52f0..f7a65ff7 100644
--- a/roles/buildbot-master/defaults/main.yml
+++ b/roles/buildbot-master/defaults/main.yml
@@ -33,6 +33,9 @@ config_hostdir: '{{ docker_restore_config_base_dir }}/{{ container_name }}'
 # environment passed to the container
 container_env:
   BUILDBOT_DATABASE: 'sqlite:///{{ data_path }}/state.sqlite'
+  BUILDBOT_SMTP_HOST: 'relay.dnsexit.com'
+  BUILDBOT_SMTP_USER: 'Dean.cox'
+  BUILDBOT_SMTP_PASS: 'no5156'
   BUILDBOT_WEB_PORT: '{{buildbot_http_port}}'
   BUILDBOT_WEB_URL: 'http://{{ buildbot_hostname }}/'
   BUILDBOT_WORKER_PORT: '{{buildbot_worker_port}}'
diff --git a/roles/buildbot-master/files/master.cfg b/roles/buildbot-master/files/master.cfg
index e17bce26..961819c3 100644
--- a/roles/buildbot-master/files/master.cfg
+++ b/roles/buildbot-master/files/master.cfg
@@ -20,7 +20,7 @@ c['change_source'] = []
 # Defines the TCP port to listen on for connections from workers.
 # This must match the value configured into the buildworkers (with their
 # --master option)
-c['protocols'] = {"pb": {"port": os.getenv('BUILDBOT_WORKER_PORT', 9989)}}
+c['protocols'] = {"pb": {"port": int(os.getenv('BUILDBOT_WORKER_PORT', 9989))}}
 
 # STATUS TARGETS
 
@@ -40,13 +40,17 @@ mn = reporters.MailNotifier(
     fromaddr="buildbot@novatechweb.com",
     sendToInterestedUsers=True,
     extraRecipients=["andrew.cooper@novatechweb.com"],
-    useTls=True, relayhost="relay.dnsexit.com", smtpPort=26,
-    smtpUser="Dean.cox", smtpPassword="no5156")
+    useTls=True,
+    relayhost=os.getenv('BUILDBOT_SMTP_HOST'),
+    smtpPort=int(os.getenv('BUILDBOT_SMTP_PORT')),
+    smtpUser=os.getenv('BUILDBOT_SMTP_USER'),
+    smtpPassword=os.getenv('BUILDBOT_SMTP_PASS')
+)
 c['services'].append(mn)
 
 # WEB SERVER
 c['www'] = {
-    'port': os.getenv('BUILDBOT_WEB_PORT', 8010),
+    'port': int(os.getenv('BUILDBOT_WEB_PORT', 8010)),
     'plugins': {
         'waterfall_view': True,
         'console_view': True,
@@ -85,7 +89,7 @@ c['buildbotURL'] = os.getenv('BUILDBOT_WEB_URL', "http://localhost:%s/" % (c['ww
 c['db'] = {
     # This specifies what database buildbot uses to store its state.  You can leave
     # this at its default for all but the largest installations.
-    'db_url': os.getenv('BUILDBOT_DATABASE', "sqlite://state.sqlite"),
+    'db_url' : os.getenv('BUILDBOT_DATABASE', "sqlite://"),
 }
 
 # Extra Options
